src/test/scala/temple/RegenerationFilterTest.scala

```diff
@@ -0,0 +1,68 @@
+package temple
+
+import java.io.{File, PrintWriter}
+
+import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+import temple.builder.project.Project
+import temple.detail.PoliceSergeantNicholasAngel
+import temple.generate.FileSystem
+
+import scala.reflect.io.Directory
+
+class RegenerationFilterTest extends FlatSpec with Matchers with BeforeAndAfter {
+
+  behavior of "RegenerationFilterTest"
+
+  private val testDirectory = "/tmp/test"
+
+  after {
+    val testDirectory = new File(this.testDirectory)
+    if (testDirectory.exists()) new Directory(testDirectory).deleteRecursively()
+  }
+
+  private def setupFileSystemWhole(): Unit = {
+    new File(testDirectory).mkdirs
+    val testFile = new File(testDirectory + "/test-file.go")
+    new PrintWriter(testFile).write("foo")
+    val setupGo = new File(testDirectory + "/setup.go")
+    new PrintWriter(setupGo).write("bar")
```

Comment:
Can we use `FileUtils.writeToFile(testDirectory + "/setup.go", "bar")`? It’s what it was created for, and closes the file for you